frontend/src/screens/apps/New.tsx

```diff
@@ -0,0 +1,231 @@
+import { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useUser } from '../../context/UserContext';
+import { BudgetRenewalType, RequestMethodType, User, UserInfo, nip47MethodDescriptions, nip47MethodIcons, validBudgetRenewals } from '../../types';
+
+const New = () => {
+  const { info } = useUser();
+  // const navigate = useNavigate();
+  const currentUser: User = (info as UserInfo).user!;
+  
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+
+  const [edit, setEdit] = useState(false);
+
+  const nameParam = (queryParams.get('name') || queryParams.get('c')) ?? ""
+  const [appName, setAppName] = useState(nameParam);
+  const pubkey = queryParams.get('pubkey') ?? "";
+  const returnTo = queryParams.get('return_to') ?? "";
+
+  const budgetRenewalParam = queryParams.get('budget_renewal') as BudgetRenewalType;
+  const budgetRenewal: BudgetRenewalType = validBudgetRenewals.includes(budgetRenewalParam) ? budgetRenewalParam : "monthly";
+
+  const reqMethodsParam = queryParams.get('request_methods');
+  const [requestMethods, setRequestMethods] = useState(reqMethodsParam ?? Object.keys(nip47MethodDescriptions).join(' '));
+
+  const maxAmountParam = queryParams.get('max_amount') ?? "";
+  const [maxAmount, setMaxAmount] = useState(parseInt(maxAmountParam));
+
+  const parseExpiresParam = (expiresParam: string): string => {
+    if (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(expiresParam)) {
+      const d = new Date(expiresParam); 
+      const isIso = d instanceof Date && !isNaN(d.getTime()) && d.toISOString() === expiresAtParam;
```

Comment:
This also looks wrong. Don't we know what type of date is returned?

Reply:
Ah, it comes from the query parameter. Maybe we just support a unix timestamp in seconds (that's what we say in the README)

Reply:
I also added support for `ISO strings` along with timestamp in seconds